src/components/Geolocation/Geolocation.js

```diff
@@ -0,0 +1,237 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import OLMap from 'ol/Map';
+import { transform } from 'ol/proj';
+import Point from 'ol/geom/Point';
+import Feature from 'ol/Feature';
+
+import Style from 'ol/style/Style';
+import Circle from 'ol/style/Circle';
+import Fill from 'ol/style/Fill';
+import Stroke from 'ol/style/Stroke';
+
+import VectorLayer from 'ol/layer/Vector';
+import VectorSource from 'ol/source/Vector';
+
+import { FaRegDotCircle } from 'react-icons/fa';
+
+import Button from '../Button';
+
+const propTypes = {
+  /**
+   * CSS class of the button.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Additional CSS class of the activated button.
+   */
+  classNameActive: PropTypes.string,
+
+  /**
+   * Title of the button.
+   */
+  title: PropTypes.string,
+
+  /**
+   * Map.
+   */
+  map: PropTypes.instanceOf(OLMap).isRequired,
+
+  /**
+   * Function triggered when geolocating is not successful.
+   */
+  onError: PropTypes.func,
+
+  /**
+   * If true, the map is not centered after it has been dragged once.
+   */
+  noCenterAfterDrag: PropTypes.bool,
+
+  /**
+   * Color (Number array with rgb values) or style function.
+   * If a color is given, the style is animated.
+   */
+  colorOrStyleFunc: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.number),
+    PropTypes.func,
+  ]),
+};
+
+const defaultProps = {
+  className: 'tm-geolocation',
+  classNameActive: 'blink',
+  title: undefined,
+  onError: () => {},
+  noCenterAfterDrag: false,
+  colorOrStyleFunc: [235, 0, 0],
+};
+
+/**
+ * This component displays a geolocator.
+ */
+class Geolocation extends PureComponent {
+  constructor(props) {
+    super(props);
+    const { map, onError, noCenterAfterDrag, colorOrStyleFunc } = this.props;
+
+    this.map = map;
+    this.onError = onError;
+    this.colorOrStyleFunc = colorOrStyleFunc;
```

Comment:
Usually we must not set class's property from props's property  otherwise if the props's property is changed by the app  the current will not be aware of this change and will always use the value set from the constructor.
That's why in class's function we always get values from props. 

I hope I'm clear :)